commit 48735d8d8bd701b1e0cd3d49c21e5e385ddcb077
Author: Johannes Berg <johannes@sipsolutions.net>
Date:   Wed Oct 29 11:43:32 2008 +0100

    libertas: fix buffer overrun
    
    If somebody sends an invalid beacon/probe response, that can trash the
    whole BSS descriptor. The descriptor is, luckily, large enough so that
    it cannot scribble past the end of it; it's well above 400 bytes long.
    
    Signed-off-by: Johannes Berg <johannes@sipsolutions.net>
    Cc: stable@kernel.org [2.6.24-2.6.27, bug present in some form since driver was added (2.6.22)]
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

Adjusted to apply to Debian's 2.6.24 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.24.orig/drivers/net/wireless/libertas/scan.c linux-source-2.6.24/drivers/net/wireless/libertas/scan.c
--- linux-source-2.6.24.orig/drivers/net/wireless/libertas/scan.c	2008-01-24 15:58:37.000000000 -0700
+++ linux-source-2.6.24/drivers/net/wireless/libertas/scan.c	2008-12-01 22:44:28.000000000 -0700
@@ -1022,8 +1022,8 @@ static int libertas_process_bss(struct b
 
 		switch (elem->id) {
 		case MFIE_TYPE_SSID:
-			bss->ssid_len = elem->len;
-			memcpy(bss->ssid, elem->data, elem->len);
+			bss->ssid_len = min_t(int, 32, elem->len);
+			memcpy(bss->ssid, elem->data, bss->ssid_len);
 			lbs_deb_scan("ssid '%s', ssid length %u\n",
 			             escape_essid(bss->ssid, bss->ssid_len),
 			             bss->ssid_len);
